nilearn/typing.py

```diff
@@ -0,0 +1,54 @@
+"""Types or Types alias used by Nilearn.
+
+
+Many of those correspond to the types of variable
+declared in nilearn._utils.doc.
+"""
+
+from __future__ import annotations
+
+import pathlib
+import sys
+from typing import Callable
+
+from joblib.memory import Memory
+from numpy import ndarray
+from numpy.typing import DTypeLike
+
+MemoryLevel = int
+NJobs = int
+Resume = bool
+Standardize = bool
+Verbose = int
+
+# TODO update when dropping python 3.9
+if sys.version_info[1] < 10:
+    DataDir = (str, pathlib.Path)
```

Comment:
from what I can tell python 3.9 isinstance is not too happy with using a type annotation like notation for its second argument

Reply:
3.9

```python
>>> isinstance(1, int | None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

3.13

```python
>>> isinstance(1, int | None)
True
```

Reply:
Depends how you see it...

python 3.9 EOL is in November :https://endoflife.date/python

so in terms of release that's 1 or 2 releases for us: pretty soon